plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergStatistics.java

```diff
@@ -266,10 +268,14 @@ private void updateMinMaxStats(
             ImmutableMap.Builder<Integer, Object> map = ImmutableMap.builder();
             idToMetricMap.forEach((id, value) -> {
                 Type.PrimitiveType type = idToTypeMapping.get(id);
-                verify(type != null, "No type for column id %s, known types: %s", id, idToTypeMapping);
-                Object icebergRepresentation = Conversions.fromByteBuffer(type, value);
-                if (icebergRepresentation != null) {
-                    map.put(id, icebergRepresentation);
+                if (type != null) {
```

Comment:
Thanks for catching this, lets actually go a little farther with the change. Can you try to:

- Inline the `Conversions.fromByteBuffer` call with the call sites in `acceptDataFile`
- Use the type from the `for (Types.NestedField column : partitionSpec.schema().columns())` loop in `acceptDataFile`  loop instead of `idToTypeMapping`
- remove `convertBounds` and `idToTypeMapping` entirely